Extract `styled-components` content

lukeed opened the pull request:
* extracts inline `<style>` rules to `document.head` 😉 
* fixes `TheButton` members in accordance to SC classname generation

The original gist/suggestions worked correctly. This PR enhances it to move those rules to the head, as per your request.

This PR also fixes/works around something that is purely `styled-components`' doing. I don't think it's even necessarily a bug – they just generate a new classname per property modifier (eg, your `[disabled]` _and_ `[opacity]` attributes) and then expect their users to redefine (or use `styled-theme` ... or manually extend) all styles for all variants. 😬 

So, the dynamic portions were always working. The problem was that your base-class identifiers were replaced with empty rules.

---

PS: I skipped precommit hooks because they didn't work on my machine. It was missing `prettier` which I assume you have installed globally

sapegin commented on src/components/TheButton.tsx:
```diff
@@ -97,7 +107,6 @@ export const TheButton = ({
 			onClick={onStart}
 			aria-label="Start timer"
 			disabled={status === 'running'}
-			opacity={status === 'running' ? 0 : 1}
```
This is a styled-system prop and already works, we shouldn't need to reimplement it manually.

sapegin commented on pwa.config.js:
```diff
@@ -1,6 +1,8 @@
+exports.babel = function (config) {
+	config.plugins.push(
+		['babel-plugin-styled-components', {
+			transpileTemplateLiterals: true,
```
I think this is the default value.

lukeed commented:
Hey,

Reverted the component changes. And added a new runtime flag for PWA that will **only** run during the prerendering phase of `pwa export`.

What was actually happening is that SC only outputs the style text for initial/base classes. Since it was built (only) for SSR, it didn't concern itself with spitting out the modifier classes. This was why `pwa watch` mode was working fine – the runtime was in full control of everything & aware of all modifier classes.

Yesterday, I was thrown off the trail because SC _does indeed_ auto-duplicate all of the base class's styles into the modifier class (eg, `.foo{color:blue}` -> `.bar{color:blue;opacity:0}`) which...no comment... but that's why it looked like the modifier code was present when it actually wasn't.

That should take care of everything :)

lukeed commented on pwa.config.js:
```diff
@@ -1,6 +1,7 @@
+exports.babel = function (config) {
+	config.plugins.push(
+		['babel-plugin-styled-components', {
+			pure: true,
```
You'll want to take these changes too @sapegin 

https://styled-components.com/docs/tooling#dead-code-elimination

sapegin replied:
Do we even use a minifier here? Who's going to read these comments?

lukeed replied:
Yes, both Babel and Terser react to the inline comment. Separately, this annotation type also works within Rollup

lukeed commented:
Closing this since you got just about everything in via https://github.com/sapegin/coffeetimer/commit/75d0f55a6408339c4e6ba59d2d4be1276de9417c

Congrats!

sapegin commented:
Thanks for help! 🤗
